File: simpleClass/R/src/UserInfo.R
UserInfo = 
  R6Class("UserInfo",
          public = list(
            
            initialize = function(user, password) {
              private$user = user
              private$password = password
            },
            
            getUser = function() { return(private$user) },
            
            getPassword = function() { 
              crypt = private$cryptPassword( private$password ) 
              return(crypt) 
            },
            
            setUser = function(user) { self$user = user },
            
            setPassword = function(password) { self$password = password }
            
          ),
          
          private = list(
            user = NULL,
            password = NULL,
            cryptPassword = function(password) {
              crypt = strsplit(password, "")[[1]]
              crypt = rev(crypt)
              return(sprintf("HASH%s000", paste(crypt, collapse = "") ))
            }
          )
 )

File: simpleClass/ruby/src/UserInfo.rb
class UserInfo

    attr_accessor :user
    attr_writer :password

    def initialize(user, password)
        @user = user
        @password = password
    end

    def password
        return self.cryptPassord(@password)
    end

    private 
    def cryptPassord(password)
        return "HASH#{password.reverse}000"
    end
end